recipes/libxml2/all/conanfile.py

```diff
@@ -299,4 +299,4 @@ def package_info(self):
         self.cpp_info.names["cmake_find_package_multi"] = "LibXml2"
         self.cpp_info.names["pkg_config"] = "libxml-2.0"
         self.cpp_info.builddirs.append(self._module_subfolder)
-        self.cpp_info.build_modules = [os.path.join(self._module_subfolder, self._module_file)]
+        self.cpp_info.build_modules["cmake_find_package"] = [self._module_file_rel_path]
```

Comment:
```suggestion
        self.cpp_info.build_modules["cmake_find_package"] = [self._module_file_rel_path]
        self.cpp_info.build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
```

Reply:
Nop

Reply:
Would something like this make sense?
https://github.com/kcat/openal-soft/blob/master/OpenALConfig.cmake.in

Reply:
I think so

Reply:
I'll update this part

Reply:
https://github.com/conan-io/conan-center-index/pull/6303/commits/5a9a249ee4092ba0719f20215fb68af0878324b9

Reply:
`bzip2`, `freetype`, `jasper`, `libgeotiff`, `openal`, `openssl`, `sqlite3`, `xapian-core` and `xz_utils` recipes could benefit from this trick.
It will be very useful when `CMakeDeps` will replace `cmake_find_package(_multi)` generators.

Reply:
It was a bad move:
https://github.com/conan-io/conan-center-index/issues/6433
https://github.com/conan-io/conan-center-index/issues/6545

I think that it should be reported to conan team, when cmake_find_package will be removed, it will cause lot of pain, and I don't think they are aware of the side effects.

Reply:
We must continue to properly emulate custom CMake variables from official Find module files provide by Kitware.
Currently, it's done with a custom module file created during packaging and added through build_modules. Emulating those variables is easy for Find module files:
For example in bzip2 recipe: https://github.com/conan-io/conan-center-index/blob/e567ce817a75db9a3822bb642845552fb920be35/recipes/bzip2/all/conanfile.py#L74-L92

But it's harder for config files, since the variables we rely on don't exist (they are suffixed with RELEASE, DEBUG etc, very fragile).

Reply:
yes, generator expressions seems to be tricky to handle...
maybe new `set_property` feature could provide mechanism to define this extra CMake variables?
e.g. I am looking at #5828 and seems like we have to provide a bunch of additional variables...
we can do it with modules, but I would love to see an easier way to do it

Reply:
It's complex, with find_package, we could delegate complexity of variables content to existing variables.

Reply:
While looking at #6433, I have gotten the impression that the cmake_find_package(_multi) generators are non-consistent, and do not have complete similar behavior when using components (or not).
This is perhaps because they are unmaintained.

`FindXXX.cmake` modules are an integral part of cmake, and it is thus a must that we are going to be able to generate them in the future.